### PR TITLE
plugin grpc Unary

### PR DESCRIPTION
**Pull request by Alipebt**

### CODE REVIEW

**Comment by wu-sheng** [comment]
You need to update docs of supported list accordingly.

**Comment by wu-sheng** [comment]
And it seems you don't add your new test in the CI files, therefore, your cases haven't been verified by CI.

File: docs/en/agent/support-plugins.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ metrics based on the tracing data.
   * `dubbo`: [Dubbo](https://github.com/apache/dubbo-go) tested v3.0.1 to v3.0.5.
   * `kratosv2`: [Kratos](https://github.com/go-kratos/kratos) tested v2.3.1 to v2.6.2.
   * `microv4`: [Go-Micro](https://github.com/go-micro/go-micro) tested v4.6.0 to v4.10.2.
+  * `grpc`: [gRPC](https://github.com/grpc/grpc-go) tested v1.56.0 to v1.56.2
```
Comment (mrproliu) [inline docs/en/agent/support-plugins.md:16]: Can you add more version checks? Because the version iteration of grpc is still very frequent.

File: plugins/grpc/client_recvmsg_interceptor.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	ctx := cs.Context()
+	method := cs.callHdr.Method
+
+	peerKey, ok := peer.FromContext(ctx)
+	remoteAddr := ""
+	if ok {
+		remoteAddr = peerKey.Addr.String()
+	}
+
+	if remoteAddr == "127.0.0.1:11800" {
```
Comment (mrproliu) [inline plugins/grpc/client_recvmsg_interceptor.go:41]: Why you need to check the this address? The grpc address is uncertainty, even send to the SkyWalking backend.

File: plugins/grpc/client_sendmsg_interceptor.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	cs := invocation.CallerInstance().(*nativeclientStream)
+	ctx := cs.Context()
+	method := cs.callHdr.Method
+	peerKey, ok := peer.FromContext(ctx)
+	remoteAddr := ""
+	if ok {
+		remoteAddr = peerKey.Addr.String()
+	}
+
+	if remoteAddr == "127.0.0.1:11800" {
```
Comment (mrproliu) [inline plugins/grpc/client_sendmsg_interceptor.go:40]: Same here, please update the checking strategy.

File: plugins/grpc/client_sendmsg_interceptor.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		return nil
+	}
+
+	s, err := tracing.CreateLocalSpan(formatOperationName(method, "/client/Request/sendMsg"),
```
Comment (mrproliu) [inline plugins/grpc/client_sendmsg_interceptor.go:44]: I think all the name should have same case rule. You can choose to capitalize the first letter of all words or not.

File: plugins/grpc/client_unary_interceptor.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/apache/skywalking-go/plugins/core/tracing"
+	"google.golang.org/grpc/metadata"
+)
+
+type ClientUnaryInterceptor struct {
+}
+
+func (h *ClientUnaryInterceptor) BeforeInvoke(invocation operator.Invocation) error {
+
+	// 获取参数
```
Comment (mrproliu) [inline plugins/grpc/client_unary_interceptor.go:32]: Please delete the Chinese comments.

File: plugins/grpc/client_unary_interceptor.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	ctx := invocation.Args()[0].(context.Context)
+	method := invocation.Args()[1].(string)
+
+	clientconn := invocation.CallerInstance().(*nativeClientConn)
+	remoteAddr := clientconn.Target()
+	if remoteAddr == "127.0.0.1:11800" {
```
Comment (mrproliu) [inline plugins/grpc/client_unary_interceptor.go:38]: Same here, please delete it.

File: plugins/grpc/server_unary_interceptor.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	"github.com/apache/skywalking-go/plugins/core/tracing"
+)
+
+type ServerUnaryInterceptor struct {
+}
+
+func (h *ServerUnaryInterceptor) BeforeInvoke(invocation operator.Invocation) error {
+	span := tracing.ActiveSpan()
+	span.Tag("RPCType", "Unary")
```
Comment (mrproliu) [inline plugins/grpc/server_unary_interceptor.go:30]: The activate span could be nil? Please add a nil check.

**Review by mrproliu — COMMENTED** [review]
Looks like the steaming related interceptor you have also commit? Please remove it and submit in other PR.

File: go.mod [hunk 8]
```diff
@@ -12,9 +12,11 @@ require (
 require (
 	github.com/golang/protobuf v1.5.3 // indirect
 	github.com/google/go-cmp v0.5.9 // indirect
-	golang.org/x/net v0.10.0 // indirect
-	golang.org/x/sys v0.8.0 // indirect
-	golang.org/x/text v0.9.0 // indirect
+	golang.org/x/mod v0.12.0 // indirect
```
Comment (mrproliu) [inline go.mod:15]: Why you need to change this file? Please try not to change the file, because it will affect the package of the user program.

File: plugins/grpc/go.mod [hunk 9]
```diff
@@ -0,0 +1,18 @@
+module github.com/apache/skywalking-go/plugins/grpc
+
+go 1.20
+
+require (
+	github.com/apache/skywalking-go/plugins/core v0.0.0-20230720062736-4bee1d47d62b
```
Comment (mrproliu) [inline plugins/grpc/go.mod:6]: We are already using the go work, so this package should not been required.

File: plugins/grpc/server_handleStream_interceptor.go [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+package grpc
+
+import (
+	"google.golang.org/grpc/metadata"
+
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	"github.com/apache/skywalking-go/plugins/core/tracing"
+)
+
+type ServerHandleStreamInterceptor struct {
```
Comment (mrproliu) [inline plugins/grpc/server_handleStream_interceptor.go:27]: Is this interceptor related to the unary request?

File: test/plugins/scenarios/grpc/go.mod [hunk 11]
```diff
@@ -0,0 +1,23 @@
+module test/plugins/scenarios/grpc
+
+go 1.20
+
+require (
+	github.com/apache/skywalking-go v0.1.0
```
Comment (mrproliu) [inline test/plugins/scenarios/grpc/go.mod:6]: Please delete this required package.

**Review by mrproliu — COMMENTED** [review]
Please update the README and support plugin list. Also, the CI seem failure, please fix.

File: plugins/grpc/server_handleStream_interceptor.go (hunk 10 above)
Comment (Alipebt) [reply plugins/grpc/server_handleStream_interceptor.go:27]: Yes, the server will call `handleStream` first whether it is for streaming or unary. `handleStream` will judge the rpc type and call related methods

**Review by mrproliu — DISMISSED** [review]
LGTM. Please fix the plugin test.

**Review by mrproliu — APPROVED** [review]
